### DISCUSSION TITLE
Excerpt/Truncate plugin

### DISCUSSION THREAD

**Discussion by harrygr** [remarkjs/remark#794] [Ideas] [answered]
It would be great to have a plugin that can truncate a markdown document to produce an excerpt which inteligently truncates the document close to the desired number of characters.

E.g say I have the following document

```md
first line

the quick brown fox jumped over the lazy dog
```

such a plugin used like `excerpt({chars: 25})` might result in

```html
<p>first line</p>
<p>the quick bro...</p>
```

There is already the [`remark-excerpt`][1] plugin but this requires there to be an excerpt identifier present (e.g. `<!-- excerpt -->`). I haven't seen anything that can inteligently produce the excerpt.

[1]: https://github.com/manovotny/remark-excerpt

**Comment by wooorm** [comment, marked as answer]
Been a while but [`rehype-infer-description-meta`](https://github.com/rehypejs/rehype-infer-description-meta) should cover this. See also `hast-util-excerpt` and `hast-util-truncate`!
